pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -152,6 +170,127 @@ def handle_text_delta(
             self._parts[part_index] = part_delta.apply(existing_text_part)
             return PartDeltaEvent(index=part_index, delta=part_delta)
 
+    def _handle_text_delta_with_thinking_tags(
+        self,
+        *,
+        vendor_part_id: VendorId | None,
+        content: str,
+        id: str | None = None,
+        thinking_tags: tuple[str, str],
+        ignore_leading_whitespace: bool = False,
+    ) -> ModelResponseStreamEvent | None:
+        """Handle text delta with thinking tag detection and buffering for split tags."""
+        start_tag, end_tag = thinking_tags
+
+        # Combine any buffered content with the new content
+        buffered = self._tag_buffer.get(vendor_part_id, '') if vendor_part_id is not None else ''
```

Comment:
What if `vendor_part_id` is `None`? Will none of this work anymore? Should we require one for this method to work?